### PR TITLE
by default signal performs simple operations on navigation axes

### PR DESCRIPTION
**Pull request by to266**
e.g. `s.sum()` now sums over navigation axes if not argument is passed

### CODE REVIEW

File: hyperspy/docstrings/signal.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
             The axis can be passed directly, or specified using the index of the
             axis in `axes_manager` or the axis name."""
 MANY_AXIS_PARAMETER = \
-    """: {int | string | axis | tuple}
+    """: {int | string | axis | tuple | None}
             Either one on its own, or many axes in a tuple can be passed. In
             both cases the axes can be passed directly, or specified using the
             index in `axes_manager` or the name of the axis. Any duplicates are
-            removed."""
+            removed. If None, navigation axes are passed"""
```
Comment (francisco-dlp) [inline hyperspy/docstrings/signal.py]: If `None`, the operation is performed over all navigation axes?

**Comment by francisco-dlp** [comment]
In #823 I forgot to ask for `User Guide` documentation. This is actually a good chance to add it. Maybe just an example with e.g. `sum` to show that the default is summing over all navigation axes and then show how to sum over one particular axis and all signal axes?

**Comment by to266** [comment]
...adding a new tests decreases coverage? any way, @francisco-dlp suggestions added

File: doc/user_guide/tools.rst [hunk 2]
```diff
@@ -240,13 +240,48 @@ The methods of this section are available to all the signals. In other chapters
 methods that are only available in specialized
 subclasses.
 
+Simple mathematical operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. versionchaned:: 0.9
```
Comment (francisco-dlp) [inline doc/user_guide/tools.rst]: Version added has no text and, therefore, it doesn't appear. What about "Passing multiple axes in a tuple."?

Comment (to266) [inline doc/user_guide/tools.rst]: I just noticed that I misspelled it - hence does not appear. "Passing.." for a title you mean?

Comment (francisco-dlp) [inline doc/user_guide/tools.rst]: No, I was meaning adding that little description inside  `versionchanged` not to confuse the reader in thinking that all that is below is new in 0.9.

Comment (to266) [inline doc/user_guide/tools.rst]: Well it's just changed, not new... Hopefully it's clear enough now (I think it is)

**Comment by francisco-dlp** [comment]
The coverage has decreased for a good reason, you only test the default `axis` value for `sum`.

**Comment by to266** [comment]
I forgot I added a new function (`amax`, which might be more obvious as `max`, as in numpy it's just an alias), hence the reduction in coverage. Should I add tests for all the other functions as well, even though they use identical 3 lines in each case?

File: hyperspy/docstrings/signal.py [hunk 3]
```diff
@@ -8,8 +8,9 @@
...
             axis in `axes_manager` or the axis name."""
 MANY_AXIS_PARAMETER = \
-    """: {int | string | axis | tuple}
+    """: {int | string | axis | tuple | None}
             Either one on its own, or many axes in a tuple can be passed. In
             both cases the axes can be passed directly, or specified using the
             index in `axes_manager` or the name of the axis. Any duplicates are
-            removed."""
+            removed. If None, the operation is performed over all navigation
+            axes."""
```
Comment (vidartf) [inline hyperspy/docstrings/signal.py]: add "(default)." at end?

**Comment by to266** [comment]
Added the "(default)", removed `amax`, as it was just an alias.

**Comment by to266** [comment]
Not sure if I do need to add all the other tests, even though the code is identical @francisco-dlp ..?

**Comment by francisco-dlp** [comment]
I think yes, even if the code is identical, they may get broken in the future individually and we won't know unless we have individual tests for them.

**Comment by to266** [comment]
Updated

**Pull request merged**